diff --git a/common/include/pcl/pcl_macros.h b/common/include/pcl/pcl_macros.h
index 96f2316..58234b0 100644
--- a/common/include/pcl/pcl_macros.h
+++ b/common/include/pcl/pcl_macros.h
@@ -96,11 +96,11 @@ namespace pcl
 #elif ANDROID
 // Use the math.h macros
 # include <math.h>
-# define pcl_isnan(x)    isnan(x)
-# define pcl_isfinite(x) isfinite(x)
-# define pcl_isinf(x)    isinf(x)
+# define pcl_isnan(x)    std::isnan(x)
+# define pcl_isfinite(x) std::isfinite(x)
+# define pcl_isinf(x)    std::isinf(x)
 
-#elif _GLIBCXX_USE_C99_MATH
+#elif _GLIBCXX_USE_C99_MATH 
 // Are the C++ cmath functions enabled?
 # include <cmath>
 # define pcl_isnan(x)    std::isnan(x)
